sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/OptimizerLoggingSuite.scala

```diff
@@ -152,4 +153,19 @@ class OptimizerLoggingSuite extends PlanTest {
       verifyLog(Level.INFO, Seq("Batch Has No Effect"))
     }
   }
+
+  test("SPARK-50329: toString for InSet should be valid for unresolved plan") {
+    withSQLConf(
+      SQLConf.PLAN_CHANGE_LOG_LEVEL.key -> "INFO"
+    ) {
+      val input = LocalRelation($"a".int, $"b".string, $"c".double)
+      val inSetPredicate = InSet($"a", Set(1, 2))
+      val query = input.select($"a", $"b").where(inSetPredicate)
+      val analyzed = query.analyze
+
+      assert(query.toString.contains("INSET (values with unresolved data types)"))
```

Comment:
What about adding the `$child` value too to the expected string:

```suggestion
      assert(query.toString.contains("'a INSET (values with unresolved data types)"))
```